docs/spec_draft.md

```diff
@@ -508,6 +509,68 @@ Numeric precision is implementation-defined.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.bitcast_convert
+
+### Semantics
+
+Performs a bitcast operation on `operand` tensor and produces a `result` tensor
+where the bits of the entire `operand` tensor are reinterpreted using the
+type of the `result` tensor.
+
+Let `E` and `E'` be the `operand` and `result` element type respectively, and
+`R = rank(operand)`:
+  * If `num_bits(E')` $=$ `num_bits(E)`,
+    `bits(result[i0, ..., iR-1]) = bits(operand[i0, ..., iR-1])`.
+  * If `num_bits(E')` $\lt$ `num_bits(E)`,
+    `bits(result[i0, ..., iR-1, :]) = bits(operand[i0, ..., iR-1])`.
+  * If `num_bits(E')` $\gt$ `num_bits(E)`,
+    `bits(result[i0, ..., iR-2]) = bits(operand[i0, ..., iR-2, :])`.
+
+The behavior of `bits` is implementation-defined because the exact
+representation of tensors is implementation-defined, and the exact
+representation of floating-point types is implementation-defined (e.g. IEEE-754
+allows implementations to deviate on endianness).
```

Comment:
Resurrecting a question from one of the resolved conversations: "Is it fair to call the behavior completely implementation-defined? Or can we guarantee something for integer to integer casts? What about booleans?".

Reply:
For conversions itself, we cannot guarantee integer to integer casts either since they are also subject to endianness. The behavior of `bits` meta function is defined for booleans since endianness is over multiple bytes, but boolean is one byte. However, this does not mean that say, float to bool or bool to floats, are defined since the floating representation is dependent on endianness. All casts (except for casts of one type to itself like bool to bool or i32 to i32) are therefore implementation-defined. We can either specify that `bits` is defined for booleans but the rest is implementation defined or just say that everything is implementation-defined.

Reply:
Sounds good. Let's just say that everything is implementation-defined.